lib/moab/storage_object.rb

```diff
@@ -153,11 +153,7 @@ def empty?
     # @api external
     # @return [Integer] The identifier of the latest version of this object, or 0 if no versions exist
     def current_version_id
-      return @current_version_id unless @current_version_id.nil?
-      #@current_version_id = self.version_id_list.last || 0
-      list = self.version_id_list
-      version_id = list.empty? ? 0 : list.last
-      @current_version_id = version_id
+      @current_version_id ||= self.version_id_list.last || 0
     end
```

Comment:
It's curious that the old code had commented out the `@current_version_id = self.version_id_list.last || 0`, as if that was considered and rejected for some reason (some bug).

Reply:
seems like this should work fine whenever the code being replaced would've worked.  if `version_id_list` is empty, `#last` will return `nil`, and `@current_version_id` will be set to `0`.  this just collapses what used to be lines 159 and 160 into one `||` statement, as well as collapsing the memoization of 157 and 161 into one `||=` statement.

unless there's some specific worry that `version_id_list` won't be a valid list, i think this is fine.

Reply:
The only problem arises if/when `self.version_id_list == nil`, then `.first` will blow up.  (I dunno why I had some spare time to look at this, but it popped up in my github notifications and I had a bit of time to look today; apologies if I'm poking in where I shouldn't).

Reply:
list.empty? will also throw error if list is nil, so no real change.

Reply:
It looks like `list.empty?` would blow up anyway if the `self.version_id_list` ever returned `nil`.

Reply:
also, `self.version_id_list` should always return a list:
https://github.com/sul-dlss/moab-versioning/blob/b2770c71d3e57515d68976ceb7aa749f81ea8e7a/lib/moab/storage_object.rb#L131-L141
```ruby
    def version_id_list
      list = Array.new
      return list unless @object_pathname.exist?
      @object_pathname.children.each do |dirname|
        vnum = dirname.basename.to_s
        if vnum =~ /^v(\d+)$/
          list << vnum[1..-1].to_i
        end
      end
      list.sort
    end
```